src/connectionconfig/connectionDialogWebViewController.ts

```diff
@@ -314,38 +293,129 @@ export class ConnectionDialogWebViewController extends ReactWebViewPanelControll
 						validationMessage: ''
 					};
 				}
-			},
-			{
-				propertyName: 'database',
-				label: 'Database',
-				required: false,
-				type: FormComponentType.Input,
-			},
-			{
-				propertyName: 'trustServerCertificate',
-				label: 'Trust Server Certificate',
-				required: false,
-				type: FormComponentType.Checkbox,
-			},
+			}
+		});
+
+		components.set('profileName', {
+			option: undefined,
+			component: {
+			propertyName: 'profileName',
+			label: 'Profile Name',
+			required: false,
+			type: FormComponentType.Input,
+			}
+		});
+
+		// add missing validation functions for generated components
+		components.get('server')!.component.validate = (value: string) => {
+			if (this.state.selectedFormTab === FormTabType.Parameters && !value) {
+				return {
+					isValid: false,
+					validationMessage: 'Server is required'
+				};
+			}
+			return {
+				isValid: true,
+				validationMessage: ''
+			};
+		};
+
+		components.get('user')!.component.validate = (value: string) => {
+			if (this.state.connectionProfile.authenticationType === AuthenticationType.SqlLogin && !value) {
+				return {
+					isValid: false,
+					validationMessage: 'User name is required'
+				};
+			}
+			return {
+				isValid: true,
+				validationMessage: ''
+			};
+		};
+
+		components.get('tenantId')!.component.validate = (value: string) => {
+			if (this.state.connectionProfile.authenticationType === AuthenticationType.AzureMFA && !value) {
+				return {
+					isValid: false,
+					validationMessage: 'Tenant ID is required'
+				};
+			}
+			return {
+				isValid: true,
+				validationMessage: ''
+			};
+		};
+	}
+
+	private _mainOptionNames = new Set<string>([
```

Comment:
consider changing this to an array to ensure consistent ordering

Reply:
Never mind:
1) JS Maps are iterated in insertion order
2) insertion order (and therefore ordering) is controlled by STS.